### PR TITLE
Hwp 369 - Create API test for Community Tags

### PR DESCRIPTION
**Pull request by mattsiel**
# Description

This PR includes the following proposed change(s):

- Jira Ticket 369
- adds tests for the community tags functions
- adds create community tests

## Requires

Add 'Requires Attention' label if appropriate.

- [ ] Requires an update to the .env file
- [ ] Requires adding a file listed in the .gitignore
- [ ] Requires an opinion or answer to a question, see comments.
- [ ] Requires other attention, see below.

If there are any requirements for the developer to make after this PR is merged please list them here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

For pull requests that are not just a simple fix, please check that the branch works on your local machine.

- [ ] Tested by Zach.
- [ ] Tested by Brandon.
- [ ] Tested by Brady.

## Does the change impact or break the Docker build?

- [ ] Yes
- [ ] No

If Yes: Has Docker been updated accordingly?

- [ ] Yes
- [ ] No

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [ ] If attention is required by the developer such as updating the .env file, these requirements have been listed.

### CODE REVIEW

File: app/server/tests/api-createCommunity.test.js [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+const welComTitle = "Welcome";
+const welComDescript = "Welcome to theNeighbourhood";
+const welComRules = "string";
+
+const newComTitle = "hello";
+const newComDescript = "world";
+const newComRules = "1. rules";
+const newComTags = [{
+    "tag": "Informative",
+    "count": 1
```
Comment (BradyMitch) [inline app/server/tests/api-createCommunity.test.js:16]: This wont be allowed in the future. Its a TODO to not allow setting of count for a tag on community creation.

File: app/server/tests/api-deleteCommunityTags.test.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+});
+
+describe('Get Communities tags', () => {
+  test('API returns the updated tag', async () => {
+    response = await community.getCommunityTags(newComTitle, token);
+    expect(" " + response.text+ " ").toContain(updatedTags);
+  });
+});
+
+// Testing the get communities function after logging in
```
Comment (BradyMitch) [inline app/server/tests/api-deleteCommunityTags.test.js:71]: Your comments don't match the functions

File: app/server/tests/api-deleteCommunityTags.test.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  });
+});
+
+
+describe('Deleting new Community', () => {
+  test('API returns a successful response - code 200', async() => {
+    let response = await community.deleteCommunity(newComTitle, token);
+    expect(response.status).toBe(200);
+  });
+});
```
Comment (BradyMitch) [inline app/server/tests/api-deleteCommunityTags.test.js:118]:
```suggestion
});

```

File: app/server/tests/api-getCommunityTags.test.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+      expect(" " + response.text+ " ").toContain("" + newComTags + "");
+    });
+});
+
+describe('Deleting new Community', () => {
+  test('API returns a successful response - code 200', async() => {
+          response = await community.deleteCommunity(newComTitle, token);
+          expect(response.status).toBe(200);
+  });
+});
```
Comment (BradyMitch) [inline app/server/tests/api-getCommunityTags.test.js:74]:
```suggestion
});

```

File: app/server/tests/api-setCommunityTags.test.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+  });
+});
+
+
+describe('Deleting new Community', () => {
+  test('API returns a successful response - code 200', async() => {
+    let response = await community.deleteCommunity(newComTitle, token);
+    expect(response.status).toBe(200);
+  });
+});
```
Comment (BradyMitch) [inline app/server/tests/api-setCommunityTags.test.js:162]:
```suggestion
});

```

File: app/server/tests/api-createCommunity.test.js (hunk 1 above)
Comment (mattsiel) [reply app/server/tests/api-createCommunity.test.js:16]: okay will switch that out

File: app/server/tests/api-deleteCommunityTags.test.js (hunk 2 above)
Comment (mattsiel) [reply app/server/tests/api-deleteCommunityTags.test.js:71]: forgot to remove the comments my bad

File: app/server/tests/api-deleteCommunityTags.test.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+describe('Creating new Community', () => {
+  test('API returns a successful response - code 201', async() => {
+    await community.deleteCommunity(newComTitle, token);
+    let response = await community.createCommunity(newComTitle, newComDescript, newComRules, newComTags, token);
+    expect(response.status).toBe(201);
+  });
+});
+
+
+// Testing the get communities function after logging in
```
Comment (mattsiel) [inline app/server/tests/api-deleteCommunityTags.test.js:55]:
```suggestion
```

File: app/server/tests/api-deleteCommunityTags.test.js [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+});
+
+describe('Get Communities tags', () => {
+  test('API returns the updated tag', async () => {
+    response = await community.getCommunityTags(newComTitle, token);
+    expect(" " + response.text+ " ").toContain(updatedTags);
+  });
+});
+
+// Testing the get communities function after logging in
```
Comment (mattsiel) [inline app/server/tests/api-deleteCommunityTags.test.js:69]:
```suggestion
```

File: app/server/tests/api-getCommunityTags.test.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+describe('Logging in the test user', () => {
+  test('API returns a successful response - code 201', async () => {
+    let response = await user.login('test2', 'Tester123!');
+    token = response.body.token;
+    expect(response.status).toBe(201);
+  });
+});
+
+
+// Testing the get communities function after logging in
```
Comment (mattsiel) [inline app/server/tests/api-getCommunityTags.test.js:27]:
```suggestion
```

File: app/server/tests/api-getCommunityTags.test.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+
+describe('Creating new Community', () => {
+    test('API returns a successful response - code 201', async() => {
+        await community.deleteCommunity(newComTitle, token);
+        response = await community.createCommunity(newComTitle, newComDescript, newComRules, newComTags, token);
+        expect(response.status).toBe(201);
+    });
+});
+
+// Testing the get communities function after logging in
```
Comment (mattsiel) [inline app/server/tests/api-getCommunityTags.test.js:51]:
```suggestion
```

File: app/server/tests/api-setCommunityTags.test.js [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+describe('Get Communities Tags - After Login with new community (2)', () => {
+  test('API returns the updated tag',  async () => {
+    response = await community.getCommunityTags(newComTitle, token);
+    expect(" " + response.text+ " ").toContain(updatedTags);
+  });
+});
+
+// Testing the get communities function after logging in
```
Comment (mattsiel) [inline app/server/tests/api-setCommunityTags.test.js:82]:
```suggestion
```

File: app/server/tests/api-createCommunity.test.js (hunk 1 above)
Comment (mattsiel) [inline app/server/tests/api-createCommunity.test.js:16]:
```suggestion
```

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**